torch/csrc/jit/codegen/cuda/python_frontend/python_bindings.cpp

```diff
@@ -187,46 +192,76 @@ void initNvFuserPythonBindings(PyObject* module) {
           py::arg("ndims"),
           py::arg("dtype") = torch::jit::fuser::cuda::DataType::Float,
           py::return_value_policy::reference)
-      .def_static(
+      .def(
           "define_tensor",
-          [](size_t ndims,
-             std::vector<bool> contiguity,
+          [](FusionDefinitionContextManager& self,
+             std::vector<int> sizes,
+             std::vector<int> strides,
              torch::jit::fuser::cuda::DataType dtype =
                  torch::jit::fuser::cuda::DataType::Float) -> TensorView* {
             TORCH_CHECK(
-                ndims == contiguity.size(),
-                "The number of dimensions specified does not match contiguity.");
-            return TensorViewBuilder()
-                .ndims(ndims)
-                .dtype(dtype)
-                .contiguity(std::move(contiguity))
-                .build();
+                sizes.size() == strides.size(),
+                "The number of sizes does not match the number of strides.",
+                sizes.size(),
+                strides.size());
+
+            std::vector<IterDomain*> domain_sizes;
+            for (const auto i : c10::irange(sizes.size())) {
+              if (sizes[i] == 1) {
+                domain_sizes.push_back(IrBuilder::create<IterDomain>(
+                    self.fusionPtr()->zeroVal(),
+                    self.fusionPtr()->oneVal(),
+                    ParallelType::Serial,
+                    IterType::BroadcastWithStride));
+              } else {
+                domain_sizes.push_back(IrBuilder::create<IterDomain>(
+                    self.fusionPtr()->zeroVal(), IrBuilder::create<Int>()));
+              }
+            }
+
+            std::vector<bool> contig_info(strides.size(), false);
+            for (int i = contig_info.size() - 1; i >= 0; --i) {
+              if (i == static_cast<int>(contig_info.size() - 1)) {
+                contig_info[i] = (strides[i] == 1);
+              } else {
+                contig_info[i] =
```

Comment:
I think we are all good here.

I'm a little uncertain how collapsing across a broadcasted dimension would be handled by codegen. But that's a separate topic we might want to investigate separately.

i.e. -> tensor with size [4, 1, 4] & stride [8, 8, 1]
I'll try that on our devel branch

Reply:
NVM.. double checked the kernel and everything looks fine.

Turns out we are not doing anything special for size-1 broadcasting, in terms of indexing... So collapsing does stop properly there.